components/Banner/examples/supplemental/MDCBannerView.m

```diff
@@ -54,124 +106,371 @@ - (instancetype)initWithCoder:(NSCoder *)aDecoder {
+      break;
+    }
+  }
+  return CGSizeMake(size.width, frameHeight);
 }
 
-- (void)layoutSubviews {
-  [super layoutSubviews];
+- (void)updateConstraints {
+  MDCBannerViewLayoutMode layoutMode = [self layoutModeForSizeToFit:self.bounds.size];
+  [self updateConstraintsWithLayoutMode:layoutMode];
 
-  [self applyMDCBannerViewLayout:self.layout];
+  [super updateConstraints];
 }
 
 #pragma mark - Layout methods
 
-- (void)applyMDCBannerViewLayout:(MDCBannerViewLayout *)layout {
-  if (![self isMDCBannerViewLayoutApplicable:layout]) {
-    return;
+- (void)updateConstraintsWithLayoutMode:(MDCBannerViewLayoutMode)layoutMode {
+  [self deactivateAllConstraints];
+  // Set Elements
+  if (@available(iOS 9.0, *)) {
+    if (layoutMode == MDCBannerViewLayoutModeSingleLine) {
+      if (self.hasImage) {
+        self.imageViewConstraintLeading.active = YES;
+        self.imageViewConstraintTopSmall.active = YES;
+        self.imageViewConstraintBottom.active = YES;
+        self.textLabelConstraintLeadingWithImage.active = YES;
+      } else {
+        self.textLabelConstraintLeadingWithMargin.active = YES;
+      }
+      self.textLabelConstraintCenterY.active = YES;
+      self.buttonContainerConstraintLeadingWithTextLabel.active = YES;
+      self.buttonContainerConstraintLeadingWithTextLabelGreater.active = YES;
+      self.buttonContainerConstraintTrailing.active = YES;
+      self.buttonContainerConstraintTopWithMargin.active = YES;
+      self.buttonContainerConstraintBottom.active = YES;
+    } else {
+      if (self.hasImage) {
+        self.imageViewConstraintLeading.active = YES;
+        self.imageViewConstraintTopLarge.active = YES;
+        self.textLabelConstraintLeadingWithImage.active = YES;
+        self.buttonContainerConstraintTopWithImageViewGreater.active = YES;
+      } else {
+        self.textLabelConstraintLeadingWithMargin.active = YES;
+      }
+      self.textLabelConstraintTop.active = YES;
+      self.textLabelConstraintTrailing.active = YES;
+      self.buttonContainerConstraintTopWithTextLabelGreater.active = YES;
+      self.buttonContainerConstraintTopWithTextLabel.active = YES;
+      self.buttonContainerConstraintLeading.active = YES;
+      self.buttonContainerConstraintTrailing.active = YES;
+      self.buttonContainerConstraintBottom.active = YES;
+    }
   }
 
-  self.iconImageViewContainer.frame = layout.imageContainerFrame;
-  self.iconImageView.center = CGPointMake(CGRectGetWidth(layout.imageContainerFrame) / 2,
-                                          CGRectGetHeight(layout.imageContainerFrame) / 2);
-  self.textLabel.frame = layout.textLabelFrame;
-  [self.buttons enumerateObjectsUsingBlock:^(id object, NSUInteger idx, BOOL *stop) {
-    if (idx >= layout.buttonFrames.count) {
-      *stop = YES;
-      return;
+  [self updateButtonsConstraintsWithLayoutMode:layoutMode];
+}
+
+#pragma mark - Layout helpers
+
+- (void)updateButtonsConstraintsWithLayoutMode:(MDCBannerViewLayoutMode)layoutMode {
```

Comment:
Consider annotating this API as `NS_AVAILABLE_IOS(9_0)` so that you can wrap calls to it in `if (@available(iOS 9.0, *)) {` instead.

Reply:
Good idea. Addressed in 0a1f2957e0cb3e9f3c11d9b7389b1f66a9b577c9.